docs/DeveloperGuide.md

```diff
@@ -1072,6 +1072,199 @@ testers are expected to do more *exploratory* testing.
+   Expected: The student at index 2 is removed from the first lesson. Confirmation message is shown.
+4. Test case: `removefromlesson 1 n/NonExistentStudent`  
+   Expected: No student is removed. Error message about student not found shown.
+5. Test case: `removefromlesson 2 n/TestOne` (where lesson at index 2 doesn’t exist)  
+   Expected: No student is removed. Error details shown.
+
+### Marking attendance in a lesson: `marka`
+
+1. Test case: `marka 1 n/TestOne a/1`  
+   Expected: Student `TestOne` is marked as attended for the first lesson. Confirmation message is shown.
+2. Test case: `marka 1 n/TestOne a/0`  
+   Expected: Student `TestOne` is marked as absent for the first lesson. Confirmation message is shown.
+3. Test case: `marka 1 n/John Doe n/Jane Doe a/1`  
+   Expected: Students `John Doe` and `Jane Doe` are both marked as attended for the first lesson. Confirmation message is shown.
+4. Test case: `marka 1 n/John Doe n/Jane Doe a/0`  
+   Expected: Students `John Doe` and `Jane Doe` are both marked as absent for the first lesson. Confirmation message is shown.
+5. Test case: `marka 1 n/NonExistentStudent a/1`  
+   Expected: No attendance is marked. Error message about student not found is shown.
+6. Test case: `marka 2 n/TestOne a/1` (where lesson at index 2 doesn’t exist)  
+   Expected: No attendance is marked. Error message about lesson not found is shown.
+
+### Marking participation in a lesson: `markp`
+
+1. Test case: `markp 1 n/TestOne pt/10`  
+   Expected: Student `TestOne` is marked with a participation score of 10 for the first lesson, and attendance is set to true. Confirmation message is shown.
+2. Test case: `markp 1 n/TestOne pt/0`  
+   Expected: Student `TestOne` is marked with a participation score of 0 for the first lesson. Attendance remains unchanged. Confirmation message is shown.
+3. Test case: `markp 1 n/John Doe n/Jane Doe pt/15`  
+   Expected: Students `John Doe` and `Jane Doe` are both marked with a participation score of 15 for the first lesson, and their attendance is set to true. Confirmation message is shown.
+4. Test case: `markp 1 n/NonExistentStudent pt/10`  
+   Expected: No participation is marked. Error message about student not found is shown.
+5. Test case: `markp 2 n/TestOne pt/10` (where lesson at index 2 doesn’t exist)  
+   Expected: No participation is marked. Error message about lesson not found is shown.
+6. Test case: `markp 1 n/TestOne pt/101`  
+   Expected: No participation is marked. Error message about invalid participation score is shown (as the score exceeds the valid range of 0-100).
+7. Test case: `markp 1 n/TestOne pt/-1`  
+   Expected: No participation is marked. Error message about invalid participation score is shown (as the score is below the valid range of 0-100).
+
+### Removing a lesson: `removelesson`
+
+1. Prerequisites: At least one lesson exists in the list.
+
+2. Test case: `removelesson 1`  
```

Comment:
similar to delete, u can delete multiple lessons at once using `deletelesson 1;2`